app/addons/databases/components.react.jsx

```diff
@@ -0,0 +1,271 @@
+        return (
+          <DatabaseRow row={item} />
+        );
+      });
+    },
+
+    render: function () {
+      var rows = this.createRows();
+      return (
+        <div className="view">
+          <table className="databases table table-striped">
+            <thead>
+              <th>Name</th>
+              <th>Size</th>
+              <th># of Docs</th>
+              <th>Update Seq</th>
+              <th>Actions</th>
+            </thead>
+            <tbody>
+            {rows}
+            </tbody>
+          </table>
+        </div>
+      );
+    }
+  });
+
+  var DatabaseRow = React.createClass({
+
+    renderGraveyard : function (row) {
+      if (row.status.isGraveYard()) {
+        return (
+          <GraveyardInfo row={row} />
+        );
+      } else {
+        return null;
+      }
+    },
+
+    render: function () {
+      var row = this.props.row;
+      var name = row.get("name");
+      var encoded = app.utils.safeURLName(name);
+      var size = Helpers.formatSize(row.status.dataSize());
+      return (
+        <tr>
+          <td>
+            <a href={"#/database/"+encoded+"/_all_docs"}>{name}</a>
+          </td>
+          <td>{size}</td>
+          <td>{row.status.numDocs()} {this.renderGraveyard(row)}</td>
+          <td>{row.status.updateSeq()}</td>
+          <td>
+            <a className="db-actions btn fonticon-replicate set-replication-start" title={"Replicate "+name} href={"#/replication/"+encoded}></a>&#160;
+            <a className="db-actions btn icon-lock set-permissions" title={"Set permissions for "+name} href={"#/database/"+encoded+"/permissions"}></a>
+          </td>
+        </tr>
+      );
+    }
+  });
+
+  var GraveyardInfo = React.createClass({
+
+    componentDidMount : function () {
+      $(this.refs.myself.getDOMNode()).tooltip();
+    },
+
+    render : function () {
+      var row = this.props.row;
+      return (
+        <i className="js-db-graveyard icon icon-exclamation-sign" ref="myself" title={"This database has just " + row.status.numDocs() + " docs and " + row.status.numDeletedDocs() + " deleted docs"}></i>
+      );
+    }
+  });
+
+  var RightDatabasesHeader = React.createClass({
+
+    render : function () {
+      return (
+        <div className="header-right">
+          <AddDatabaseWidget />
+          <JumpToDatabaseWidget />
+        </div>
+      );
+    }
+  });
+
+  var AddDatabaseWidget = React.createClass({
+
+    onTrayToggle : function () {
+      var that = this;
+      this.refs.newDbTray.toggle(function (shown) {
+        if (shown) {
+          that.refs.newDbName.getDOMNode().focus();
+        }
+      });
+    },
+
+    onKeyUpInInput : function (e) {
+      if (e.which === 13) {
+        this.onAddDatabase();
+      }
+    },
+
+    onAddDatabase : function () {
+      var databaseName = $(this.refs.newDbName.getDOMNode()).val();
+      Actions.createNewDatabase(databaseName, this.refs.newDbTray.hide, FauxtonAPI.addNotification);
```

Comment:
I'm with you with FauxtonAPI.addNotification
However, I'd keep the hide callback: using another action dispatch just to hide the tray is a lot of overhead - and it gets more brittle: 1.) right now you have one tray, but for how long? 2.) When passing in, you already KNOW which react node you're dealing with, you don't have to remember for an event coming in and 3.) you might not want to change behavior - which right now is: hide the tray in the middle of things (when all trivial checks were performed and going back is possible).
So, I'd leave the callback here (plus: testing is also quite simple)